pkg/server/tls/dynamic_source_test.go

```diff
@@ -296,7 +296,7 @@ func TestDynamicSource_FailingSign(t *testing.T) {
 			}
 
 			// Start the DynamicSource
-			ctx, cancel := context.WithCancel(context.Background())
+			ctx, cancel := context.WithCancel(context.WithoutCancel(t.Context()))
```

Comment:
This looks a bit odd. Could you explain why this must be like this? Maybe a comment in code?

Reply:
good catch, I updated the PR and simplified this.